### PR TITLE
[Pimcore 11] Decouple AdminBundle from the Core

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Follow up to https://github.com/pimcore/pimcore/pull/14918

### <samp>🤖 Generated by Copilot at d463847</samp>

This pull request refactors the admin UI to use a new framework and removes the legacy admin UI bundle. It updates the ApplicationLoggerBundle and the CustomReportsBundle to be compatible with the new framework and the new security configuration. It also changes the WebDAV endpoint from `/admin/asset/webdav` to `/asset/webdav` and renames some files and routes accordingly.

### <samp>🤖 Generated by Copilot at d463847</samp>

> _`pimcore_admin` gone_
> _New admin UI framework_
> _Spring of refactoring_

### <samp>🤖 Generated by Copilot at d463847</samp>

*  Remove the pimcore_admin.config_location configuration and the PimcoreAdminBundle, as they are no longer needed for the new admin UI framework ([link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-159a57553c090df4264e28ad7cd3df02a4817e814970a81ed2a06c1016f8f008L44-R42), [link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-740cfa12da3718779279532c3de8317ccea826dcd94ab08dc5d3b5aa67e81875L34))
*  Rename the pimcore_admin_webdav firewall and route to pimcore_webdav and move the WebDAV endpoint to /asset/webdav instead of /admin/asset/webdav, as part of the refactoring of the admin UI ([link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-ee8ff688fded8943af83197fb3b0953bf27ea9ff986d6754f5c4d1a5b5767a3eL14-R18), [link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-162565d9ee75d98a6d1fc62d5b530627454cc7d15e480cf2c398570ebef7a006L35-R39), [link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-ef84abe3e2623ebd0de534a7bc4fe8df32fa6a07d39e194079dc06af1225c174L35-R35))
*  Remove the composer requirement for the pimcore/admin-ui-classic-bundle, which is a legacy bundle that provides the old admin UI and is deprecated ([link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-5cc0a3dbbc060d0b5c0785dd3549a0f96ce106526737164f607243be2ad295b5L19-L20))
*  Remove the import of the default.yaml file from the ApplicationLoggerBundle and CustomReportsBundle config, as it is now imported by the admin-classic.yaml file, which is only loaded if the pimcore_admin extension is present ([link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-88d00174f8bceb1715f8d91d40efd83bed0ca54f0276ba26f73afc639dadd0dbL1-L3), [link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-70f45cbf3abcdd056480b02ba7a0f54aef315f93b4b1af998fcf87dd552381d3L1-L3))
*  Move the ApplicationLoggerBundle and CustomReportsBundle config and service definitions from the pimcore/config.yaml and services.yaml files to the admin-classic.yaml file, which is only loaded if the pimcore_admin extension is present, to avoid conflicts with the new admin UI framework ([link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-88d00174f8bceb1715f8d91d40efd83bed0ca54f0276ba26f73afc639dadd0dbL12-R9), [link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-2f138e412c9272f5e87d7fd0dea24eefdddebca024c84c44ab51f4bebdeb1a4dL15-R24), [link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-0255e97b6da4d34073143f860d353c59cfec1affcbebe7c5e27bf610cf1eb8e7L16-L18))
*  Update the use statements, controller classes, and methods of the ApplicationLoggerBundle and CustomReportsBundle to make them compatible with the new admin UI framework and the new JsonResponse class ([link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-fa008ad5c77b53fa6cfd4553b3a68894f98b8d98c1aac7958d363005480be48aL22-R25), [link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-fa008ad5c77b53fa6cfd4553b3a68894f98b8d98c1aac7958d363005480be48aL38-R44), [link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-fa008ad5c77b53fa6cfd4553b3a68894f98b8d98c1aac7958d363005480be48aL63-R78), [link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-fa008ad5c77b53fa6cfd4553b3a68894f98b8d98c1aac7958d363005480be48aL152-R157), [link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-fa008ad5c77b53fa6cfd4553b3a68894f98b8d98c1aac7958d363005480be48aL194-R199), [link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-fa008ad5c77b53fa6cfd4553b3a68894f98b8d98c1aac7958d363005480be48aL213-R218))
*  Implement the PrependExtensionInterface and add the prepend method to the PimcoreApplicationLoggerExtension class of the ApplicationLoggerBundle, to allow the extension to load the admin-classic.yaml file if the pimcore_admin extension is present ([link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-7a5207a1d0a41a7ad40379eac6885011d7f8bc707b06927c392853123fa0a435L20-R25), [link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-7a5207a1d0a41a7ad40379eac6885011d7f8bc707b06927c392853123fa0a435R36-R47))
*  Replace the PimcoreBundleAdminSupportInterface and BundleAdminSupportTrait with the PimcoreBundleAdminClassicInterface and BundleAdminClassicTrait in the PimcoreApplicationLoggerBundle class of the ApplicationLoggerBundle, to make the bundle compatible with the new admin UI framework and the new admin bundle interface and trait ([link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-e5a231dc9e8ace8744aeacd24d628997133aea262bf85b839a97a2ff299a715fL19-R26))
*  Add the import of the security.yaml file to the CoreBundle config, which contains the security configuration for the new admin UI framework ([link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-aaa126d697a44a4f20dc63a638d6a2aafbebc2c0278c8cf619975135d470b468L4-R5))
*  Rename the authenticator_security.yaml file to security.yaml in the CoreBundle config ([link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-fd854ba5be2eb4b4d3341e4141cdff84ba4ad04f1cb2695da704e44fd152abbf))
*  Make the Pimcore\Security\User\UserProvider service public and add the services for the notification model, which are used by the new admin UI framework ([link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-d8b32ac5dfd96e61b04a0acc639e673c92a989df9281e4a3d36ae9e5b07a50d9L156-R157), [link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-d8b32ac5dfd96e61b04a0acc639e673c92a989df9281e4a3d36ae9e5b07a50d9R174-R183))
*  Add the void return type to the process method of the SerializerPass class of the CoreBundle, as it is required by the parent interface ([link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-cfab4d00b02b3634d20bd929e1e82abb412fe8de738281244cb1bd62d6c90d89L38-R38))
*  Rename the default.yaml file to admin-classic.yaml in the ApplicationLoggerBundle config ([link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-18194583815f66b82b20ae371d9f428cc3671cf30a88ec1b75bd356531fdc6a6))
*  Delete the AdminAbstractController class and the authenticator_security.yaml file from the AdminBundle, as they are replaced by the new admin UI framework ([link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-ef2da637a288f08af5cbae98b4704f0a84c62cbd3e5c7919d59891f186830dbe), [link](https://github.com/pimcore/pimcore/pull/14971/files?diff=unified&w=0#diff-f10314085fd72d3c206913a198477709872319fee3c1e984a8b892798168aa8c))

### CODE REVIEW

File: tests/Support/Helper/Pimcore.php [hunk 1]
```diff
@@ -286,6 +308,13 @@ public function _before(TestInterface $test): void
 
         $this->groups = $test->getMetadata()->getGroups();
 
+        $path = TestHelper::resolveFilePath('system_settings.json');
+        if (!file_exists($path)) {
+            throw new \RuntimeException(sprintf('System settings file in %s was not found', $path));
+        }
+        $data = file_get_contents($path);
+        SettingsStore::set('system_settings', $data, 'string', 'pimcore_system_settings');
+
```
Comment (dvesh3) [inline tests/Support/Helper/Pimcore.php]:
```suggestion
```

File: .github/ci/files/config/packages/security.yaml [hunk 2]
```diff
@@ -11,40 +11,11 @@ security:
             security: false
 
         # Pimcore WebDAV HTTP basic // DO NOT CHANGE!
-        pimcore_admin_webdav:
-            pattern: ^/admin/asset/webdav
+        pimcore_webdav:
+            pattern: ^/asset/webdav
```
Comment (dvesh3) [inline .github/ci/files/config/packages/security.yaml:15]: Route needs to be changed as well along with upgrade notes and doc changes.

File: .github/ci/files/src/Kernel.php [hunk 3]
```diff
@@ -31,7 +31,7 @@ class Kernel extends BaseKernel
     public function registerBundlesToCollection(BundleCollection $collection): void
     {
         // pimcore bundles
-        $collection->addBundle(new PimcoreAdminBundle(), 60);
+        //$collection->addBundle(new PimcoreAdminBundle(), 60);
```
Comment (dvesh3) [inline .github/ci/files/src/Kernel.php]: cleanup

File: .github/ci/scripts/setup-pimcore-environment.sh [hunk 4]
```diff
@@ -17,4 +17,4 @@ if [ ${PIMCORE_STORAGE:-local} = "minio" ]; then
     composer require -n --no-update league/flysystem-aws-s3-v3
 fi
 
-composer require -n --no-update pimcore/admin-ui-classic-bundle:"@dev"
+#composer require -n --no-update pimcore/admin-ui-classic-bundle:"dev-decouple_from_core as 1.0.0"
```
Comment (dvesh3) [inline .github/ci/scripts/setup-pimcore-environment.sh]: cleanup

File: bundles/SimpleBackendSearchBundle/src/Controller/SearchController.php [hunk 5]
```diff
@@ -348,7 +358,12 @@ public function findAction(Request $request, EventDispatcherInterface $eventDisp
         $eventDispatcher->dispatch($afterListLoadEvent, AdminSearchEvents::SEARCH_LIST_AFTER_LIST_LOAD);
         $result = $afterListLoadEvent->getArgument('list');
 
-        return $this->adminJson($result);
+        return $this->jsonResponse($result);
+    }
+
+    protected function extractSortingSettings(array $params): array
+    {
+        return QueryParams::extractSortingSettings($params);
```
Comment (dvesh3) [inline bundles/SimpleBackendSearchBundle/src/Controller/SearchController.php:374]: check for `QueryParams` class exists

File: bundles/WordExportBundle/src/Controller/TranslationController.php [hunk 6]
```diff
@@ -34,8 +35,10 @@
  * @Route("/translation")
  *
  */
-class TranslationController extends AdminController
+class TranslationController extends UserAwareController
 {
+    use JsonHelperTrait;
```
Comment (dvesh3) [inline bundles/WordExportBundle/src/Controller/TranslationController.php:40]: not used anywhere..

File: models/Element/AbstractElement.php [hunk 7]
```diff
@@ -468,7 +468,7 @@ public function getUserPermissions(?User $user = null): array
 
         foreach ($permissions as $type => $isAllowed) {
             $event = new ElementEvent($this, ['isAllowed' => $isAllowed, 'permissionType' => $type, 'user' => $user]);
-            \Pimcore::getEventDispatcher()->dispatch($event, AdminEvents::ELEMENT_PERMISSION_IS_ALLOWED);
+            \Pimcore::getEventDispatcher()->dispatch($event, ElementEvents::ELEMENT_PERMISSION_IS_ALLOWED);
```
Comment (dvesh3) [inline models/Element/AbstractElement.php:471]: Add upgrade notes

File: phpstan-baseline.neon [hunk 8]
```diff
@@ -63,3 +173,18 @@ parameters:
 			message: "#^Class Pimcore\\\\Bundle\\\\PersonalizationBundle\\\\Targeting\\\\Document\\\\DocumentTargetingConfigurator not found\\.$#"
 			count: 1
 			path: models/Document/Editable/Snippet.php
+
+		-
+			message: "#^Call to static method get\\(\\) on an unknown class Pimcore\\\\Bundle\\\\AdminBundle\\\\CustomView\\\\Config\\.$#"
```
Comment (dvesh3) [inline phpstan-baseline.neon]: Remove from core as this moved to admin bundle https://github.com/pimcore/admin-ui-classic-bundle/pull/29/files#diff-dcae69140ecbd1a845e9be5e0e5eac41522f163308adda099b940a5aa0e7866fR30

File: bundles/WordExportBundle/src/Controller/TranslationController.php (hunk 6 above)
Comment (dvesh3) [reply bundles/WordExportBundle/src/Controller/TranslationController.php:40]: calling `decodeJson`

File: lib/Messenger/Handler/CleanupThumbnailsHandler.php [hunk 9]
```diff
@@ -35,6 +35,7 @@ public function __invoke(CleanupThumbnailsMessage $message, Acknowledger $ack =
         return $this->handle($message, $ack);
     }
 
+    // @phpstan-ignore-next-line
```
Comment (mattamon) [inline lib/Messenger/Handler/CleanupThumbnailsHandler.php:38]: whats the problem here?

Comment (dvesh3) [reply lib/Messenger/Handler/CleanupThumbnailsHandler.php:38]: It complains about the method not being used anywhere. Infact, It is used by messenger batch handler trait.

**Review by mattamon — COMMENTED** [review]
Rest looks now fine to me.

**Review by mattamon — APPROVED** [review]
LGTM

**Pull request merged**